src/Illuminate/Http/Resources/Json/Resource.php

```diff
@@ -91,12 +91,12 @@ public function resolve($request = null)
             $request = $request ?: Container::getInstance()->make('request')
         );
 
-        if (is_array($data)) {
-            $data = $data;
-        } elseif ($data instanceof Arrayable || $data instanceof Collection) {
-            $data = $data->toArray();
-        } elseif ($data instanceof JsonSerializable) {
-            $data = $data->jsonSerialize();
+        if (! is_array($data)) {
```

Comment:
Can't we just remove this statement completely?

Reply:
Wouldn't that be a breaking change? I kept it because it matched the behaviour before removing it.

Reply:
How would it break anything? The other two checks don't match arrays

Reply:
I know..but didn't want to risk it incase there is any edge case.

Reply:
There's no cases where an instance of `Collection`, `Arrayable` or `JsonSerializable` will pass `is_array`

Reply:
i know.. but these things often have some weird edge cases :laughing: